tests/test_logger.py

```diff
@@ -24,3 +24,19 @@ def test_bind_log_prefix():
         assert messages[1]["footest.b"] == 2
         assert messages[1]["customprefix.bar"] == "baz"
         assert messages[1]["footest.a"] == 1
+
+
+# When WOODCHIPPER_KEY_PREFIX is set, ensure calling clear_prefix() allows subsequent
+# logger calls are output without a prefix
```

Comment:
I'm going to look at this again tomorrow with fresh eyes, but if anyone looks before me: this test passes when run individually but fails when run after the other test in the same file due to empty JSON input. Things I've tried so far: closing the buffer from the first test at the end of it, renaming all the variables (which should be irrelevant due to scoping anyway), and breaking the logger into a pytest fixture with a function scope. The latter causes the first test to fail rather than the second, in the same way (no JSON output). I also pulled both tests into different files. They always pass when separated like that, event if run one after the other. I'm thinking it's something about re-initializing the same logger?

Reply:
The solution would be to only call configure once or to call get_logger() with different values.

Reply:
That all makes sense. Yesterday after I left the comment above I also ran into [this section in structlog's docs](https://www.structlog.org/en/stable/performance.html#:~:text=Later%20calls%20of%20configure()%20don%E2%80%99t%20have%20any%20effect%20on%20already%20cached%20loggers%20%E2%80%93%20that%20shouldn%E2%80%99t%20matter%20outside%20of%20testing%20though.) -- it explains that when `cache_logger_on_first_use` is True, any subsequent applications of .configure() don't have any affect. I'll try both of the options you suggested and see what happens. Thank you!

Reply:
Ahhh but remember that woodchipper.configure and structlog.configure are different. Woodchipper.configure also calls `logging.config.dictConfig` which _will_ change your logger even if its cached.

Reply:
Calling "get_logger" with different values did the trick. Going to test one more scenario with upstream locally then I'll get a new version pushed

Reply:
See my comment on our private Slack.